app/post_test.go

```diff
@@ -668,6 +668,35 @@ func TestCreatePost(t *testing.T) {
 		require.Nil(t, err)
 		assert.Equal(t, "![image]("+proxiedImageURL+")", rpost.Message)
 	})
+
+	t.Run("Sets POST_PROPS_MENTION_HIGHLIGHT_DISABLED and sends an ephemeral post when user does not have required permission", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+
+		th.AddUserToChannel(th.BasicUser, th.BasicChannel)
+
+		postWithNoMention := &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "This post does not have mentions",
+			UserId:    th.BasicUser.Id,
+		}
+		rpost, err := th.App.CreatePost(postWithNoMention, th.BasicChannel, false)
+		require.Nil(t, err)
+		assert.Equal(t, rpost.Props, model.StringInterface{})
+
+		th.RemovePermissionFromRole(model.PERMISSION_USE_CHANNEL_MENTIONS.Id, model.CHANNEL_USER_ROLE_ID)
+
+		postWithMention := &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "This post has @here mention @all",
+			UserId:    th.BasicUser.Id,
+		}
+		rpost, err = th.App.CreatePost(postWithMention, th.BasicChannel, false)
+		require.Nil(t, err)
+		assert.Equal(t, rpost.Props[model.POST_PROPS_MENTION_HIGHLIGHT_DISABLED], true)
+
+		th.AddPermissionToRole(model.PERMISSION_USE_CHANNEL_MENTIONS.Id, model.CHANNEL_USER_ROLE_ID)
+	})
```

Comment:
Is it worth testing "does not set the prop" like the one for the patch?

Reply:
I checked `does not set the prop` in the same test at line 683 above
https://github.com/mattermost/mattermost-server/pull/13902/files#diff-d6eac3439f7fb1aaa867e73e2cb2e498R683-R685

Reply:
2 and 3 are tested but 1 and 4 are not.

||HasMention|HasPermission|HasProp|
|--|--|--|--|
|1|T|T|F|
|2|T|F|T|
|3|F|T|F|
|4|F|F|F|

For a sanity check—to ensure that nobody breaks this in the future—it might be worth adding a testing them.

Reply:
You are right @mkraft  - I have added tests for those two cases. I dont think I need any more tests now though since there is pretty extensive unit tests on all the functions being used in the newly added block of code